feat: TID API 작성

minikryc opened the pull request:
### PR 타입(하나 이상의 PR 타입을 선택해주세요)
- [x] 기능 추가
- [ ] 기능 삭제
- [ ] 버그 수정
- [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트

### 반영 브랜치
local -> feat

### 변경 사항
TID 기능을 추가했습니다.
TID 관련 entity, controller, service를 생성했고, 다음과 같은 API를 구현했습니다.
- TID 질문 불러오기
- TID 답안 제출하기
- TID 답안 불러오기

### 테스트 결과
테스트를 해보지 못했습니다ㅜㅠ

minikryc committed:
tid api

minikryc committed:
tid api

eogns47 commented on src/main/java/OrangeCorps/LBridge/Entity/TIDQuestion.java:
```diff
@@ -0,0 +1,22 @@
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Getter
+    @Setter
```
getter와 setter를 questionId 에만 적용한 이유가 있나요??

eogns47 commented on src/main/java/OrangeCorps/LBridge/Entity/TIDAnswer.java:
```diff
@@ -0,0 +1,54 @@
+
+@Entity
+@Getter
+@Setter
```
Entity에 Setter를 쓰는 것보다는 DTO를 사용하는 것을 권장드려요
해당 글 참고하시면 좋을 것 같습니다
https://thenicesj.tistory.com/301

eogns47 commented on src/main/java/OrangeCorps/LBridge/Service/TIDImplementation.java:
```diff
@@ -0,0 +1,39 @@
+import java.util.List;
+
+@Service
+public class TIDImplementation {
```
TidService를 상속받고 override 하는게 의도에 적합할 것 같아요

eogns47 commented on src/main/java/OrangeCorps/LBridge/Controller/TIDController.java:
```diff
@@ -0,0 +1,55 @@
+
+    // TID 답안 확인
+    @GetMapping("/{answerId}")
+    public ResponseEntity<TIDAnswer> getAnswer(@PathVariable Long answerId) {
```
answer Id를 DB가 생성하는 것으로 보이는데 사용자가 파라미터에 넣을 때 알 수 있는 방법이 있나요? 
제가 코드를 잘 이해한건지 모르겠네요

eogns47 commented on src/main/java/OrangeCorps/LBridge/Controller/TIDController.java:
```diff
@@ -0,0 +1,55 @@
+            @PathVariable Long questionId,
+            @RequestBody Map<String, String> answerData,
+            @RequestHeader(name = "Authorization") String token) {
+        tidService.createAnswer(questionId, answerData.get("answer"), token);
```
tidServicce를 상속받아 구현한 클래스가 없어서 실행이 안될 것 같아요

eogns47 commented on src/main/java/OrangeCorps/LBridge/Entity/TIDQuestion.java:
```diff
@@ -0,0 +1,22 @@
+    private Long questionId;
+    private String category;
+    private String question;
+    private LocalDateTime timestamp;
```
TID에 타임스탬프를 넣으신 이유가 뭔가요??

minikryc committed:
TID Api 에러 수정, TID 답변 조회 엔드포인트 수정

eogns47 commented on src/main/java/OrangeCorps/LBridge/Controller/TIDController.java:
```diff
@@ -47,9 +47,9 @@ public ResponseEntity<String> createAnswer(
-    @GetMapping("/{answerId}")
-    public ResponseEntity<TIDAnswer> getAnswer(@PathVariable Long answerId) {
-        TIDAnswer answer = tidService.getAnswer(answerId);
+    @GetMapping("/{questionId}")
```
questionId도 answerId와 같은 맥락으로 알 방법이 없지 않나요 ??

minikryc committed:
- TIDAnswer entity에 userId, coupleId, combinedKey (datestamp + userid1 + userid2) attribute 추가.
- TIDAnswerRepository에서 combinedKey (datestamp + userid1 + userid2)로 조회하도록 findByCombinedKey로 수정.
- TIDController에서 TID 답안제출, 답안 확인 추가된 entity attributes 반영 수정.
- TIDService와 TIDImplementation에서 createAnswer 패러미터 추가, getAnswer 패러미터 combinedKey로 수정.

eogns47 commented:
고생했어!! 👍
